gradoop-flink/src/main/java/org/gradoop/flink/datagen/foodbroker/FoodBroker.java

```diff
@@ -0,0 +1,272 @@
+    // used for type hinting when loading vertex data
+    TypeInformation<Vertex> vertexTypeInfo = TypeExtractor
+      .createTypeInfo(gradoopFlinkConfig.getVertexFactory().getType());
+    // used for type hinting when loading edge data
+    TypeInformation<Edge> edgeTypeInfo = TypeExtractor
+      .createTypeInfo(gradoopFlinkConfig.getEdgeFactory().getType());
+
+    // Phase 1: Create MasterData
+    initMasterData();
+
+    // Phase 2.1: Run Brokerage
+    DataSet<Long> caseSeeds = env.generateSequence(1, foodBrokerConfig
+      .getCaseCount());
+
+    DataSet<GraphTransaction> brokerage = caseSeeds
+      .mapPartition(new Brokerage(gradoopFlinkConfig.getGraphHeadFactory(), gradoopFlinkConfig
+        .getVertexFactory(), gradoopFlinkConfig.getEdgeFactory(), foodBrokerConfig))
+      .withBroadcastSet(customerDataMap, Constants.CUSTOMER_MAP_BC)
+      .withBroadcastSet(vendorDataMap, Constants.VENDOR_MAP_BC)
+      .withBroadcastSet(logisticsDataMap, Constants.LOGISTIC_MAP_BC)
+      .withBroadcastSet(employeesDataMap, Constants.EMPLOYEE_MAP_BC)
+      .withBroadcastSet(productsQualityDataMap, Constants.PRODUCT_QUALITY_MAP_BC)
+      .withBroadcastSet(productsPriceDataMap, Constants.PRODUCT_PRICE_MAP_BC);
+
+    long complaintSeed = foodBrokerConfig.getCaseCount();
+
+    // Phase 2.2: Run Complaint Handling
+    DataSet<Tuple2<GraphTransaction, Set<Vertex>>> complaintHandlingTuple = brokerage
+      .flatMap(new RelevantElementsFromBrokerage())
+      .flatMap(new ComplaintHandling(
+        gradoopFlinkConfig.getGraphHeadFactory(),
+        gradoopFlinkConfig.getVertexFactory(),
+        gradoopFlinkConfig.getEdgeFactory(), foodBrokerConfig, complaintSeed))
+      .withBroadcastSet(customerDataMap, Constants.CUSTOMER_MAP_BC)
+      .withBroadcastSet(vendorDataMap, Constants.VENDOR_MAP_BC)
+      .withBroadcastSet(logisticsDataMap, Constants.LOGISTIC_MAP_BC)
+      .withBroadcastSet(employeesDataMap, Constants.EMPLOYEE_MAP_BC)
+      .withBroadcastSet(productsQualityDataMap, Constants.PRODUCT_QUALITY_MAP_BC)
+      .withBroadcastSet(employees, Employee.CLASS_NAME)
+      .withBroadcastSet(customers, Customer.CLASS_NAME);
+
+    DataSet<GraphTransaction> complaintHandling = complaintHandlingTuple
+      .map(new Value0Of2<GraphTransaction, Set<Vertex>>());
+
+    // Phase 3: combine all data
+    DataSet<Tuple3<GraphHead, Set<Vertex>, Set<Edge>>> transactionTriple = brokerage
+      .union(complaintHandling)
+      .map(new GraphTransactionTriple());
+
+    DataSet<Vertex> transactionalVertices = transactionTriple
+      .flatMap(new TransactionVertices())
+      .returns(vertexTypeInfo);
+
+    DataSet<Edge> transactionalEdges = transactionTriple
+      .flatMap(new TransactionEdges())
+      .returns(edgeTypeInfo);
+
+    DataSet<GraphHead> graphHeads = transactionTriple
+      .map(new TransactionGraphHead())
+      .returns(graphHeadTypeInfo);
+
+    DataSet<Map<GradoopId, GradoopIdList>> graphIds = transactionalEdges
```

Comment:
inefficient. please first union masterdata and perform update in a join function